### PR TITLE
Prove J

### PR DESCRIPTION
**Pull request by jonsterling**

### CODE REVIEW

File: test/success/J.prl [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+   [a : ty]
+   [fam : (-> [x : ty] (path [_] ty a x) (U #l kan))]
+   [d : ($ fam a (abs [_] a))]
+   [x : ty]
+   [p : (path [_] ty a x)]
+   ($ fam x p))
+] by [
+  lam ty, a, fam, d, x, p.
+    `(coe 0~>1
+      [i] ($ fam (@ p i) (@ ($ (Connection #l) ty a x p) i))
```
Comment (favonia) [inline test/success/J.prl]: I believe you do not need connection, and can prove this by
```
($ fam (whatever to make this type check) (@ ($ (a hcom from refl with one side being refl and the other side being p) ty a x p) i))
```

Comment (jonsterling) [reply test/success/J.prl]: Cool! I'll try your idea

Comment (jonsterling) [reply test/success/J.prl]: Hmm, I couldn't figure out how to make it work, at least not immediately.

Comment (favonia) [reply test/success/J.prl]: Something like this? Don't really want to switch branches now.
```
Thm J(#l:lvl) : [
  (->
   [ty : (U #l kan)]
   [a : ty]
   [fam : (-> [x : ty] (path [_] ty a x) (U #l kan))]
   [d : ($ fam a (abs [_] a))]
   [x : ty]
   [p : (path [_] ty a x)]
   ($ fam x p))
] by [
  lam ty, a, fam, d, x, p.
    `(coe 0~>1
      [i] ($ fam (hcom 0~>1 ty a [i=0 [_] a] [i=1 [j] (@ p j)])
                 (abs [j] (hcom 0~>j ty a [i=0 [_] a] [i=1 [j] (@ p j)])))
      d)
].
```

Comment (jonsterling) [reply test/success/J.prl]: @favonia Wow, nicely done! My composition-fu is not yet at that level 😄

File: test/success/J.prl [hunk 2]
```diff
@@ -0,0 +1,28 @@
+Def J/square(#i:dim,#j:dim, #ty, #a, #p) = [
+  (hcom 0~>#j #ty #a
+   [#i=0 [_] #a]
+   [#i=1 [j] (@ #p j)])
+].
+
+Print Welp.
```
Comment (favonia) [inline test/success/J.prl]: It's strange to me that MLton on travis is okay with this... `Welp` is not defined.

Comment (jonsterling) [reply test/success/J.prl]: @favonia Hmm, that's interesting... I would like to track that down.

File: script/test-mlton.sh [hunk 3]
```diff
@@ -4,4 +4,4 @@ echo "Building RedPRL with MLton..."
 ./script/mlton.sh >build.log 2>&1 || { echo "build failed!"; cat build.log; exit 1; }
 echo "done!"
 
-# exec ./script/test-no-build.sh
+exec ./script/test-no-build.sh
```
Comment (jonsterling) [inline script/test-mlton.sh]: @favonia This is what happened

**Pull request merged**